src/scripts/tray.ts

```diff
@@ -1,10 +1,40 @@
 import { BrowserWindow, Tray } from "electron";
+import { existsSync } from "fs";
+import { settings } from "../constants/settings";
+import { settingsStore } from "./settings";
 import { getMenu } from "./menu";
 
 let tray: Tray;
 
+/**
+ * Get the icon path to use for the tray icon
+ * Checks if custom path is set and valid, otherwise uses default
+ * @param defaultIcon the default bundled icon path
+ * @returns the icon path to use
+ */
+function getTrayIconPath(defaultIcon: string): string {
+  const customPath = settingsStore.get<string>(settings.trayIconPath);
```

Comment:
the type paramater `string` isn't enough, it only sets the `key` type whereas we're mostly interested in the `value` type (since we need it to be able to do things like `.trim()`;